Mathlib/Analysis/NormedSpace/Connected.lean

```diff
@@ -122,6 +124,47 @@ section NormedSpace
 
 variable {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
 
+section Ball
+
+namespace Metric
+
+theorem ball_contractible {x : E} {r : ℝ} (hr : 0 < r) :
+    ContractibleSpace (ball x r) :=
+  Convex.contractibleSpace (convex_ball _ _) (by simpa)
+
+theorem eball_contractible {x : E} {r : ENNReal} (hr : 0 < r) :
+    ContractibleSpace (EMetric.ball x r) := by
+  cases r with
+  | top =>
+    rw [eball_top_eq_univ, (Homeomorph.Set.univ E).contractibleSpace_iff]
+    exact RealTopologicalVectorSpace.contractibleSpace
+  | coe r =>
+    rw [emetric_ball_nnreal]
+    apply ball_contractible
+    simpa using hr
+
+theorem ball_pathConnectedSpace {x : E} {r : ℝ} (hr : 0 < r) :
+    PathConnectedSpace (ball x r) :=
+  @ContractibleSpace.instPathConnectedSpace _ _ (ball_contractible hr)
+
+theorem eball_pathConnectedSpace {x : E} {r : ENNReal} (hr : 0 < r) :
+    PathConnectedSpace (EMetric.ball x r) :=
+  @ContractibleSpace.instPathConnectedSpace _ _ (eball_contractible hr)
```

Comment:
I think this should be using `IsPathConnected`, not `PathConnectedSpace`, because these are sets.